Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/SteamEngine.cs

```diff
@@ -0,0 +1,1037 @@
+        /// </summary>
+        public float LPCompPressure_m_AtmPSI;
+
+        /// <summary>
+        /// Cylinder Pressures
+        /// </summary>
+        public float LPCompPressure_n_AtmPSI;
+
+        /// <summary>
+        /// Cylinder Pressures
+        /// </summary>
+        public float LPCompPressure_q_AtmPSI;
+
+        /// <summary>
+        /// Cylinder Pressures
+        /// </summary>
+        public float Pressure_a_AtmPSI;
+
+        /// <summary>
+        /// Cylinder Pressures
+        /// </summary>
+        public float Pressure_b_AtmPSI;
+
+        /// <summary>
+        /// Cylinder Pressures
+        /// </summary>
+        public float Pressure_c_AtmPSI;
+
+        /// <summary>
+        /// Initial Pressure to LP cylinder @ start if stroke
+        /// </summary>
+        public float LPPressure_a_AtmPSI;
+
+        /// <summary>
+        /// Pressure in combined HP & LP Cylinder pre-cutoff
+        /// </summary>
+        public float LPPressure_b_AtmPSI;
+
+        /// <summary>
+        /// Pressure in LP cylinder when steam release valve opens
+        /// </summary>
+        public float LPPressure_c_AtmPSI;
+
+        /// <summary>
+        /// Back pressure on LP cylinder
+        /// </summary>
+        public float LPPressure_d_AtmPSI;
+
+        /// <summary>
+        /// Pressure in LP cylinder when exhaust valve closes, and compression commences
+        /// </summary>
+        public float LPPressure_e_AtmPSI;
+
+        /// <summary>
+        /// Pressure in LP cylinder after compression occurs and steam admission starts
+        /// </summary>
+        public float LPPressure_f_AtmPSI;
+
+        /// <summary>
+        /// Force out of wheel axle model after slip
+        /// </summary>
+        public float CompensatedAxleForceN;
+
+        /// <summary>
+        /// Previous axle speed
+        /// </summary>
+        public float previousAxleSpeedMpS;
+
+
+        /// <summary>
+        /// Mean Effective Pressure
+        /// </summary>
+        public float MeanEffectivePressurePSI;
+
+        /// <summary>
+        /// Steam usage per steam engine
+        /// </summary>
+        public float CylinderSteamUsageLBpS;
+
+        /// <summary>
+        /// Indicated Horse Power
+        /// </summary>
+        public float IndicatedHorsePowerHP;
+
+        /// <summary>
+        /// Calculated Factor of Adhesion
+        /// </summary>
+        public float CalculatedFactorOfAdhesion;
+
+        /// <summary>
+        /// Maximum Tractice Effort
+        /// </summary>
+        public float MaxTractiveEffortLbf;
+
+        /// <summary>
+        /// Static wheel force for engine
+        /// </summary>
+        public float SteamStaticWheelForce;
+
+        /// <summary>
+        /// Axle weight for engine
+        /// </summary>
+        public float AxleWeightN;
+
+        /// <summary>
+        /// wheel slip for engine
+        /// </summary>
+        public bool WheelSlip;
```

Comment:
You should remove the following variables from this class:
WheelSlip, AxleWeightN, previousAxleSpeedMpS, CompensatedAxleForceN, DriveWheelRadiusM and DriveWheelWeightKg, because all these are already defined in LocomotiveAxle class.
You also have to move NumberDriveWheelAxles from this class to LocomotiveAxle class, so diesel locomotives with coupled wheels use this same parameter.

Then, you have to add a reference here to a LocomotiveAxle, which represents the axle to which this engine is coupled. For example, you can do this.Axle = Locomotive.LocomotiveAxles[this.Id];